pyproject.toml

```diff
@@ -3,6 +3,126 @@ name = "openpilot"
 requires-python = ">= 3.11"
 readme = "README.md"
 license = {text = "MIT License"}
+version = "0.1.0"
+description = "an open source driver assistance system"
+authors = ["Vehicle Researcher <user@comma.ai>"]
+dependencies = [
+
+# multiple users
+"sounddevice",  # micd + soundd
+"pyserial",     # pigeond + qcomgpsd
+"requests",     # many one-off uses
+"sympy",        # rednose + friends
+"crcmod",       # cars + qcomgpsd
+"tqdm",         # cars (fw_versions.py) on start + many one-off uses
+
+# hardwared
+"smbus2",  # configuring amp
+
+# core
+"cffi",
+"scons",
+"pycapnp",
+"Cython",
+"setuptools",
+"numpy < 2.0.0",
+
+# body / webrtcd
+"aiohttp",
+"aiortc",
+"pyaudio",
+
+# panda
+"libusb1",
+"spidev; platform_system == 'Linux'",
+
+# modeld
+"onnx >= 1.14.0",
+"onnxruntime >=1.16.3; platform_system == 'Linux' and platform_machine == 'aarch64'",
+"onnxruntime-gpu >=1.16.3; platform_system == 'Linux' and platform_machine == 'x86_64'",
+
+# logging
+"pyzmq",
+"sentry-sdk",
+
+# athena
+"PyJWT",
+"json-rpc",
+"websocket_client",
+
+# acados deps
+"casadi",
+"future-fstrings",
+
+# these should be removed
+"psutil",
+"pycryptodome",  # used in updated/casync, panda, body, and a test
+
+#logreader
+"zstd",
+]
+
+[project.optional-dependencies]
+docs = [
+  "Jinja2",
+  "sphinx",
+  "sphinx-rtd-theme",
+  "sphinx-sitemap"
+]
+
+testing = [
+  "coverage",
+  "hypothesis ==6.47.*",
+  "mypy",
+  "pre-commit",
+  "pytest",
+  "pytest-cov",
+  "pytest-cpp",
+  "pytest-subtests",
+  "pytest-xdist",
+  "pytest-timeout",
+  "pytest-randomly",
+  "pytest-asyncio",
+  "pytest-mock",
+  "pytest-repeat",
+  "ruff"
+]
+
+dev = [
+"av",
+"azure-identity",
+"azure-storage-blob",
+"breathe",
+"control",
+"dictdiffer",
+"flaky",
+"inputs",
+"lru-dict",
+"matplotlib",
+"metadrive-simulator; platform_machine != 'aarch64'",
+"mpld3",
+"myst-parser",
+"natsort",
+"opencv-python-headless",
+"parameterized >=0.8, <0.9",
+#pprofile = "*"
+"pyautogui",
+"pygame < 2.6.0", # TODO: unpin when they release a prebuilt wheel
+"pyopencl; platform_machine != 'aarch64'",  # broken on arm64
+"pywinctl",
+"pyprof2calltree",
+"rerun-sdk",
+"tabulate",
+"types-requests",
+"types-tabulate",
+
+# this is only pinned since 5.15.11 is broken
+"pyqt5 ==5.15.2; platform_machine == 'x86_64'", # no aarch64 wheels for macOS/linux
+
+]
+
+[tool.uv.sources]
```

Comment:
there's no way to do this generically? does `pip install .` without uv still work?

Reply:
Yeah, once I add back the [build-system] table, `pip install .` will work and install all the core dependencies from dependencies = [...]

for the [tool.uv.sources] table, uv does not allow a github source + markers (platform_machine), you have to split it into 2: the github source in [tool.uv.sources] and all the version/platform requirements in dependencies=[...]. This is why metadrive is split in 2 places